src/pages/onboarding/StudyInfoPage.tsx

```diff
@@ -18,10 +31,37 @@ const StudyInfoPage = () => {
   const term = useUserStore((state) => state.term);
   const setTerm = useUserStore((state) => state.setTerm);
 
+  const lastStudiedUnit = useUserStore((state) => state.lastStudiedUnit);
+  const setLastStudiedUnit = useUserStore((state) => state.setLastStudiedUnit);
+
   const studyTime = useUserStore((state) => state.studyTime);
   const setStudyTime = useUserStore((state) => state.setStudyTime);
 
   const navigate = useNavigate();
+
+  const handleGradeChange = (displayValue: string) => {
+    const storeValue =
+      gradeDisplayToStore[displayValue as keyof typeof gradeDisplayToStore];
+    setGrade(storeValue);
+  };
+
+  const displayGrade =
+    gradeStoreToDisplay[grade as keyof typeof gradeStoreToDisplay];
+
+  const unitOptions = useMemo(() => {
+    const currentUnitsData =
+      subjectUnits[grade as keyof typeof subjectUnits] || [];
+
+    const options: string[] = [];
+    currentUnitsData.map((subjectData) => {
+      subjectData.units.map((unitName) => {
+        options.push(`${subjectData.subject} - ${unitName}`);
+      });
+    });
```

Comment:
@xeoxxn 

`Array.map`은 본래 반한값을 기반으로 새 배열을 만들어주는 함수이기 때문에 options를 빈 배열로 할당한 뒤, push하는 형태보단 한 줄로 사용하는 것이 좀 더 깔끔하고 목적에 맞게 함수를 사용하는 것 같아요! 아래처럼 변경해보는 건 어떨까요? 지금은 `forEach`의 성격이 강한 것 같아서요.

```suggestion
    const options = currentUnitsData.flatMap((subjectData) =>
      subjectData.units.map((unitName) => `${subjectData.subject} - ${unitName}`)
    );
```